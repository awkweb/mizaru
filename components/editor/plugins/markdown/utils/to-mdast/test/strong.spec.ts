import toMDAST from '../'

for (const syntax of ['**', '__']) {
    describe(syntax, () => {
        test('single word', () => {
            const mdast = toMDAST(`${syntax}foo${syntax}`)
            expect(mdast).toMatchSnapshot()
        })

        test('multiple words', () => {
            const mdast = toMDAST(`${syntax}foo bar baz${syntax}`)
            expect(mdast).toMatchSnapshot()
        })

        test('split across multiple lines', () => {
            const mdast = toMDAST(`${syntax}foo\nbar\nbaz${syntax}`)
            expect(mdast).toMatchSnapshot()
        })

        describe('nested', () => {
            test('emphasis', () => {
                const mdast = toMDAST(`${syntax}foo *bar* baz${syntax}`)
                expect(mdast).toMatchSnapshot()
            })

            test('strong', () => {
                const mdast = toMDAST(`${syntax}foo **bar** baz${syntax}`)
                expect(mdast).toMatchSnapshot()
            })
        })
    })
}
