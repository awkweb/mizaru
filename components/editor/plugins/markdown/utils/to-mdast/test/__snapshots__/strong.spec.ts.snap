// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`** multiple words 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "syntax",
              "value": "**",
            },
            Object {
              "type": "text",
              "value": "foo bar baz",
            },
            Object {
              "type": "syntax",
              "value": "**",
            },
          ],
          "type": "strong",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`** nested emphasis 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "syntax",
              "value": "**",
            },
            Object {
              "type": "text",
              "value": "foo ",
            },
            Object {
              "children": Array [
                Object {
                  "type": "syntax",
                  "value": "*",
                },
                Object {
                  "type": "text",
                  "value": "bar",
                },
                Object {
                  "type": "syntax",
                  "value": "*",
                },
              ],
              "type": "emphasis",
            },
            Object {
              "type": "text",
              "value": " baz",
            },
            Object {
              "type": "syntax",
              "value": "**",
            },
          ],
          "type": "strong",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`** nested strong 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "syntax",
              "value": "**",
            },
            Object {
              "type": "text",
              "value": "foo ",
            },
            Object {
              "type": "syntax",
              "value": "**",
            },
          ],
          "type": "strong",
        },
        Object {
          "type": "text",
          "value": "bar",
        },
        Object {
          "children": Array [
            Object {
              "type": "syntax",
              "value": "**",
            },
            Object {
              "type": "text",
              "value": " baz",
            },
            Object {
              "type": "syntax",
              "value": "**",
            },
          ],
          "type": "strong",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`** single word 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "syntax",
              "value": "**",
            },
            Object {
              "type": "text",
              "value": "foo",
            },
            Object {
              "type": "syntax",
              "value": "**",
            },
          ],
          "type": "strong",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`** split across multiple lines 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "syntax",
              "value": "**",
            },
            Object {
              "type": "text",
              "value": "foo",
            },
            Object {
              "type": "lineFeed",
            },
            Object {
              "type": "text",
              "value": "bar",
            },
            Object {
              "type": "lineFeed",
            },
            Object {
              "type": "text",
              "value": "baz",
            },
            Object {
              "type": "syntax",
              "value": "**",
            },
          ],
          "type": "strong",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`__ multiple words 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "syntax",
              "value": "__",
            },
            Object {
              "type": "text",
              "value": "foo bar baz",
            },
            Object {
              "type": "syntax",
              "value": "__",
            },
          ],
          "type": "strong",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`__ nested emphasis 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "syntax",
              "value": "__",
            },
            Object {
              "type": "text",
              "value": "foo ",
            },
            Object {
              "children": Array [
                Object {
                  "type": "syntax",
                  "value": "*",
                },
                Object {
                  "type": "text",
                  "value": "bar",
                },
                Object {
                  "type": "syntax",
                  "value": "*",
                },
              ],
              "type": "emphasis",
            },
            Object {
              "type": "text",
              "value": " baz",
            },
            Object {
              "type": "syntax",
              "value": "__",
            },
          ],
          "type": "strong",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`__ nested strong 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "syntax",
              "value": "__",
            },
            Object {
              "type": "text",
              "value": "foo ",
            },
            Object {
              "children": Array [
                Object {
                  "type": "syntax",
                  "value": "**",
                },
                Object {
                  "type": "text",
                  "value": "bar",
                },
                Object {
                  "type": "syntax",
                  "value": "**",
                },
              ],
              "type": "strong",
            },
            Object {
              "type": "text",
              "value": " baz",
            },
            Object {
              "type": "syntax",
              "value": "__",
            },
          ],
          "type": "strong",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`__ single word 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "syntax",
              "value": "__",
            },
            Object {
              "type": "text",
              "value": "foo",
            },
            Object {
              "type": "syntax",
              "value": "__",
            },
          ],
          "type": "strong",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`__ split across multiple lines 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "syntax",
              "value": "__",
            },
            Object {
              "type": "text",
              "value": "foo",
            },
            Object {
              "type": "lineFeed",
            },
            Object {
              "type": "text",
              "value": "bar",
            },
            Object {
              "type": "lineFeed",
            },
            Object {
              "type": "text",
              "value": "baz",
            },
            Object {
              "type": "syntax",
              "value": "__",
            },
          ],
          "type": "strong",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;
